presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -189,12 +185,21 @@ public boolean isLimitGuaranteed(ConnectorSession session)
             return cachedTableHandle;
         }
         Optional<JdbcTableHandle> tableHandle = delegate.getTableHandle(session, schemaTableName);
-        if (tableHandle.isPresent() || cacheMissing) {
+        if ((tableHandle.isPresent() && canBeCached(tableHandle.get())) || (tableHandle.isEmpty() && cacheMissing)) {
             tableHandleCache.put(key, tableHandle);
         }
         return tableHandle;
     }
 
+    private static boolean canBeCached(JdbcTableHandle tableHandle)
+    {
+        // Cannot cache JdbcColumnHandle objects, since their Type may depend on session properties
+        return tableHandle.getGroupingSets().isEmpty() &&
+                tableHandle.getColumns().isEmpty() &&
```

Comment:
nit: could you please put `&&` on the next line as it is more related to next line that to current.

Reply:
i think i am following code style here.